internal/controlplane/handlers_authz.go

```diff
@@ -530,32 +497,70 @@ func (s *Server) removeInvite(
 	role authz.Role,
 	email string,
 ) (*minder.RemoveRoleResponse, error) {
-	prj := targetPrj.String()
-	// Get all invitations for this email, project and role
-	inviteToRemove, err := s.store.GetInvitationByEmailAndProjectAndRole(ctx, db.GetInvitationByEmailAndProjectAndRoleParams{
+	// Get all invitations for this email and project
+	invitesToRemove, err := s.store.GetInvitationsByEmailAndProject(ctx, db.GetInvitationsByEmailAndProjectParams{
 		Email:   email,
 		Project: targetPrj,
-		Role:    role.String(),
 	})
 	if err != nil {
-		if errors.Is(err, sql.ErrNoRows) {
-			return nil, util.UserVisibleError(codes.NotFound, "no invitation found for this email, project and role")
-		}
 		return nil, status.Errorf(codes.Internal, "error getting invitation: %v", err)
 	}
 
+	// If there are no invitations for this email, return an error
+	if len(invitesToRemove) == 0 {
+		return nil, util.UserVisibleError(codes.NotFound, "no invitations found for this email and project")
+	}
+
+	// Find the invitation to remove. There should be only one invitation for the given role and email in the project.
+	var inviteToRemove *db.GetInvitationsByEmailAndProjectRow
+	for _, i := range invitesToRemove {
+		if i.Role == role.String() {
+			inviteToRemove = &i
+			break
+		}
+	}
+	// If there's no invitation to remove, return an error
+	if inviteToRemove == nil {
+		return nil, util.UserVisibleError(codes.NotFound, "no invitation found for this role and email in the project")
+	}
 	// Delete the invitation
-	_, err = s.store.DeleteInvitation(ctx, inviteToRemove.Code)
+	ret, err := s.store.DeleteInvitation(ctx, inviteToRemove.Code)
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "error deleting invitation: %v", err)
 	}
 
+	// Resolve the project's display name
+	prj, err := s.store.GetProjectByID(ctx, ret.Project)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get project: %s", err)
+	}
+
+	// Get the sponsor's user information (current user)
+	sponsorUser, err := s.store.GetUserByID(ctx, ret.Sponsor)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get user: %s", err)
+	}
+
+	// Resolve the sponsor's identity and display name
+	identity, err := s.idClient.Resolve(ctx, sponsorUser.IdentitySubject)
+	if err != nil {
+		zerolog.Ctx(ctx).Error().Err(err).Msg("error resolving identity")
+		return nil, util.UserVisibleError(codes.NotFound, "could not find identity %q", sponsorUser.IdentitySubject)
```

Comment:
Does this need to be a fatal error? (Ditto for getting the sponsorUser above) - could we just return some raw ID? Or do you think it's better to just return all well-formed as the UI expects it or fail?

Reply:
Addressed in https://github.com/stacklok/minder/pull/3710